README.md

```diff
@@ -1,2 +1,215 @@
 # redux-phunk
-A redux middleware to make your actions phunktional.
+[![npm version](https://img.shields.io/npm/v/redux-phunk.svg?style=flat-square)](https://www.npmjs.org/package/single-spa)
+[![Build Status](https://img.shields.io/travis/CanopyTax/redux-phunk.svg?style=flat-square)](https://travis-ci.org/CanopyTax/single-spa)
+
+A redux [middleware](http://redux.js.org/docs/advanced/Middleware.html) to encourage smarter reducers and more phunktional action creators.
+
+## Motivation
+Reducers are the best place to put conditional logic and data transformations, but many redux middleware patterns encourage putting that stuff into action creators. By moving logic out of action creators and into reducers, the logic is more reusable and testable. Redux-phunk makes your action creators just smart enough to handle async actions like API calls, but keeps them dumb enough so that conditional logic and data transformations remain in reducers.
+
+## What does it do?
+Redux-phunk allows you to dispatch an array of actions that will be executed in sequence. For asynchronous code such as ajax, a special type of object (called a [`phunk`](#phunks-thats-not-even-a-word)) is included in the array, which all subsequent actions in the sequence will wait for.
+
+When you dispatch an array of actions, the [return value will be a Promise](#knowing-when-a-sequence-is-done) that resolves when all of the actions in the array have been dispatched.
+
+```js
+import { asyncValue } from 'redux-phunk';
+
+function retrieveWeather() {
+  return [
+    {
+      type: "GETTING_WEATHER",
+    },
+    {
+      name: "currentWeather",
+      async phunk() {
+        return await fetch('/api/weather')
+          .then(response => response.json());
+      }
+    },
+    {
+      type: "GOT_WEATHER",
+      weather: asyncValue("currentWeather"),
+    }
+  ];
+}
+
+store.dispatch(retrieveWeather()).then(() => {
+  console.log('done dispatching sequence!')
+});
+```
+
+## Installation
+```bash
+npm install --save redux-phunk
+```
+Then use [applyMiddleware](http://redux.js.org/docs/api/applyMiddleware.html) to use redux-phunk.
+```js
+import reducer from './reducer.js';
+import { createStore, applyMiddleware } from 'redux';
+import phunk from 'redux-phunk';
+
+const store = createStore(reducer, applyMiddleware(phunk));
+
+store.dispatch(myAction());
+```
+
+## Phunks?? That's not even a word!
+A "phunk" is just an object with `name` and `phunk` properties. The `phunk` property is a function that returns an asynchronous value, such as a Promise or a (single-valued) Observable. Examples:
```

Comment:
Maybe add a little more clarity with the differences between a `phunk` and an `action`. Do `phunks` get dispatched?